fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/AccessHints.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * AccessHints.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.query.plan.cascades;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Represents a set of AccessHint a query or a match candidate has.
+ */
+public class AccessHints {
```

Comment:
This class looks a bit superfluous to me since it appears to only wrap a `Set` of `AccessHint`s, can we get rid of it and just use a `Set<AccessHint>` instead?

Reply:
so currently `AccessHints` contains 1 logic `containsAll`, it is a little different than a traditional `Set.containsAll` (a query contains no hint, implies that it can be matched with any index), so it might be clearer to extract it as a class or at least a method. We can potentially put it in `AccessHint` class as well, I don't have a strong preference here.